src/worker.cc

```diff
@@ -1160,8 +1172,10 @@ void service_worker_t::enqueue_statistics(Api& api) const {
 
     statsd_client->count(action + ".info" + worker + ".ok", 1, 1.f, statsd_client->tags);
   }
+#endif // ENABLE_STATSD_CLIENT
 }
 midgard::Finally<std::function<void()>> service_worker_t::measure_scope_time(Api& api) const {
+#ifdef ENABLE_STATSD_CLIENT
```

Comment:
i would suggest a different fix here.. while i dont disagree that disabling the statsd client is a useful thing to have. the main thin you are trying to accomplis here is to remove the need of using `typeid`. indeed this function below actually doesnt make any calls to statsd at all yet it is disabled. in fact this function adds statistics to the protobuf which we later emit through statsd but i think these stats being in protobuf is still useful regardless of sending to statsd. thought i cant blame you for thinking of them as tightly coupled.

also, statsd is disabled so long as you dont configure it. if your statsd host is either a bogus one or a blank string (the default) the client is disabled. we still make a function call but it returns immediately.

anyway i propose a different approach than this one. i propose you simple refactor the workers to know their own names.

basically add a pure virtual method to `std::string service_worker_t::name() const = 0` this will force you to implement that m ethod in all the service workers and then you can call that in the fucntions below and not have to use `typeid` at all. i think this will be simpler and not require the `ifdef` swiss cheese